### PR TITLE
Project 18: Mongo API

### PR DESCRIPTION
**Pull request by Thessan**
[https://thessans-mongodb-book-api.herokuapp.com/](https://thessans-mongodb-book-api.herokuapp.com/)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3,35 +3,98 @@ import bodyParser from 'body-parser'
...
+// Endpoint/route that return a single book based on the ID
+  app.get('/books/book/:bookID', async (request, response) => {
+    try {
+      const specificBook = await Book.findOne({ bookID: request.params.bookID })
+      
+        if (specificBook) {
+        response.json(specificBook)
+        
+      } else {
+        response.status(404).json({ error: "Sorry, book not found" })
```
Comment (JHarde) [inline server.js:70]: Good use of showing the status of the page! 👍

File: server.js [hunk 2]
```diff
@@ -3,35 +3,98 @@ import bodyParser from 'body-parser'
...
+      }
+    } catch (err) {
+      res.status(400).json({ error: "Sorry, invalid Book ID" })
+    }
```
Comment (JHarde) [inline server.js:74]: And clear error message as well! :)

File: server.js [hunk 3]
```diff
@@ -3,35 +3,98 @@ import bodyParser from 'body-parser'
...
+      response.status(404).json("Sorry, no books found by that author");
+    }
+    response.json(authorBooks);
+  });
+
+
+  // Endpoint/route that return top rated books with a rating over 4
+  app.get('/books/top-rated', async (request, response) => {
+    const topRatedBooks = await Book.find({ average_rating: { $gte: 4 } })
+
```
Comment (JHarde) [inline server.js:93]: I like that you've sorted the books based on rating, great job! 🎖️

File: server.js [hunk 4]
```diff
@@ -3,35 +3,98 @@ import bodyParser from 'body-parser'
...
+// Endpoint/route that return all of the books in the data 
+app.get('/books', async (request, response) => {
+  const allBooks = await Book.find()
+
+    response.json(allBooks)
 })
 
+// Endpoint/route that return a single book based on the ID
+  app.get('/books/book/:bookID', async (request, response) => {
```
Comment (JHarde) [inline server.js:62]: Good job with the endpoints and restful routes!

**Review by JHarde — APPROVED** [review]
Great job Therése! It's clear that you understand how endpoints and restful routes work and I really like that you've added error messages and status in your code! 👍

**Review by KarolCislowski — APPROVED** [review]
Great woork 🥇

**Pull request closed without merging**